.buildkite/metricbeat/pipeline.yml

```
@@ -1,5 +1,44 @@
 # yaml-language-server: $schema=https://raw.githubusercontent.com/buildkite/pipeline-schema/main/schema.json
+env:
+  IMAGE_UBUNTU_X86_64: "family/core-ubuntu-2204"
+  IMAGE_UBUNTU_ARM_64: "core-ubuntu-2004-aarch64"
+  IMAGE_WIN_2016: "family/core-windows-2016"
+  IMAGE_WIN_2019: "family/core-windows-2019"
+  IMAGE_WIN_2022: "family/core-windows-2022"
+  IMAGE_MACOS_X86_64: "generic-13-ventura-x64"
+  BEATS_PROJECT_NAME: "metricbeat"
 
 steps:
-  - label: "Example test"
-    command: echo "Hello!"
+
+  - input: "Input Parameters"
+    key: "input-run-all-stages"
+    fields:
+    - select: "Metricbeat - runAllStages"
+      key: "runAllStages"
+      options:
+        - label: "True"
+          value: "true"
+        - label: "False"
+          value: "false"
+      default: "false"
+    - select: "Metricbeat - runMacOsTests"
+      key: "UI_MACOS_TESTS"
+      options:
+        - label: "True"
+          value: "true"
+        - label: "False"
+          value: "false"
+      default: "false"
+    if: "build.source == 'ui'"
+
+  - wait: ~
+    if: "build.source == 'ui'"
+    allow_dependency_failure: false
+
+  - label: ":linux: Check steps conditions"
+    key: "check-steps-conditions"
+    command: ".buildkite/scripts/check_conditions.sh"
```

I think the name could be more specific, something like

```suggestion
  - label: ":linux: Load dynamic metricbeat pipeline"
    key: "metricbeat-pipeline"
    command: ".buildkite/scripts/generate_metricbeat_pipeline.sh"
```